### ISSUE TITLE
Wakey wakey chocolate cakey

### ISSUE THREAD

**Issue by jhjacobs81** [linuxkettle/AppShop#7]
;-) i have a matrix server running still ;) 

and im thinking about building something Debian based.. so a nice appstore would be nice ;-)

**Comment by kettle-7** [comment]
Delicious. I've been considering installing Debian since about a year and a half ago, before I knew what FreeBSD was, and while I was scared of taking the plunge into Linux.

The whole thing of "linux is only for programmers" is total shit.

**Comment by jhjacobs81** [comment]
i would love to use BSD, but im no programmer, and i think no system compares to Apple just yet.

So the best alternative, for the short term, is using Debian with vanilla Gnome, bundled with all the Gnome apps that adhere to the Gnome guidelines. I would then use the AppShop to publish good quality apps. and build a distro around that :)

untill such time that i know enough of a programming language to write my own display manager / desktop environment that can match the beauty of Apple's. 

At one time i even tried working with Apple's opensource Darwin kernel, but its not usable yet :( and i'm afraid it never will. 

So now i'm wondering.. should i really learn Go, or should i focus on something like Vala? can i build a desktop environment with that? 

SSOOOO many questions..

**Comment by kettle-7** [comment]
> should i really learn Go, or should i focus on something like Vala?

If you learn Vala, then we are at least speaking the same language :~)

Vala is the language elementary OS uses, it's definitely capable of making a desktop environment, (I did it), but something Qt-based might be better, otherwise Unity would have used GTK.

Actually, I made a PyQt-based desktop environment, which doesn't look great but that's my doing. It looks like Windows 95. Then there's the PyQt-based display manager I'm working on, you could fork that, but I'll probably convert it to QML.

You could use QML for it, anything that needs C++ code I'm okay with too.

The other thing, you're not going to like this, is forking Haiku. Take the path Apple chose not to!

I would like to build a distro, I did it with TWin as the desktop environment once. GLib wouldn't compile because dhcp client was borken. I decided it would be best if I based it on Arch, so in the end I did. 'Twas then that my hard drive failed.

**Comment by kettle-7** [comment]
I've probably just made more questions than answers...

**Comment by jhjacobs81** [comment]
Nope, just more idea’s!

so, we’re gonna build a new os then? Based on the strong foundations of QTML??

i have no idea what it is..

do you jave any examples to show? Screenshots or anything?

**Comment by jhjacobs81** [comment]
But first! Get on Matrix! I made a server for you 🙈

**Comment by jhjacobs81** [comment]
matrix.swimmingturtle.eu ;-)

**Comment by jhjacobs81** [comment]
I didnt know BeOS went on in Haiku, but from what i see it lacks applications, and of ourse a modern GUI 🙈

**Comment by jhjacobs81** [comment]
But DO tell me about your desktop environment.. :) can i omplememt it on Debian? Can we work on it togetjer?? I learn quickly..

**Comment by kettle-7** [comment]
> so, we’re gonna build a new os then? Based on the strong foundations of QTML??

I guess so :~)

> from what i see it lacks applications, and of ourse a modern GUI

Nah, it has plenty of apps in the HaikuDepot store. The only thing I couldn't find was a good svg editor - Inkscape needs gtk which takes forever to compile and hasn't been packed.

> DO tell me about your desktop environment.. :) can i omplememt it on Debian?

You should be able to, but probably won't want to. It really is that bad. Despite the fact I used it as my only desktop for weeks. It worked on Ubuntu. It depended on a few MATE components (caja and marco, the filer and window manager being the only ones I remember) and also the `python3-psutil` package for battery applet. It needs a 1366x768 screen, it will be dysfunctional on anything smaller and on anything bigger it'll sit halfway down the screen. QML would fix that.

> Can we work on it togetjer?

I'd make an entirely new Wayland QML compositor before returning to "kettleDE". I would have used KDE but the name was taken.

**Comment by jhjacobs81** [comment]
okay yes, so lets connect on matrix ;) makes for easier discussions! 

i looked into QML while i was supposed to be sleeping, but it's javascript based? ive seen some tutorials that didnt look too hard :)

**Comment by kettle-7** [comment]
Yeah, that's why I like using it. If you want a button that resizes to half the width of what ever it's inside, all you need is `width: parent.width / 2`.

> lets connect on matrix

I tried with element, I need a username or room name to connect to.

**Comment by jhjacobs81** [comment]
Let me see if registering is enabled :)

**Comment by kettle-7** [comment]
> vanilla Gnome, bundled with all the Gnome apps that adhere to the Gnome guidelines.

Well you're about to be amazed:

https://www.omgubuntu.co.uk/2021/07/adwaita-borderless-theme-in-development-gnome-41

**Comment by kettle-7** [comment]
I'm not updating GTK any time soon...

**Comment by jhjacobs81** [comment]
Neither am i? ;-)

**Comment by kettle-7** [comment]
As far as distro goes, I think the likes of Arch would be good to fork. Debian maybe too, but there's a lot going on in that boot process.

And after playing around with Qt Wayland Compositor, we could have a splash screen with something interesting while the system boots, then transition that into the display manager, like winlogon.exe does.

**Comment by jhjacobs81** [comment]
These are good idea’s :) as for the distro.. i couldn’t care less ;-) as long as we can make it rock solid, and pretty ;-)

**Comment by kettle-7** [comment]
> rock solid

They're both pretty stable, but Debian uses old versions of software: people don't get the latest versions of apps they want, but decoupling from dpkg could be good for apps: using app bundles or AppImages.

**Comment by jhjacobs81** [comment]
And we would bundle AppShop ;-)

**Comment by jhjacobs81** [comment]
The truth is, i am willing to change a distro (although ive been using Debian since 1996) if it gets me something that could rival macOS :)

**Comment by kettle-7** [comment]
> ive been using Debian since 1996

I haven't stuck at a distro for more than three months, helloSystem might have something to say about that...

**Comment by jhjacobs81** [comment]
It’ll be a while before HS is stable and usable for “the common folk” ;-)

**Comment by kettle-7** [comment]
Same could be said for Gentoo Linux, but plenty of people use it anyway :~)

Chrome OS comes to mind...

**Comment by jhjacobs81** [comment]
If onky Chrome wasnt a google shiz..

**Comment by kettle-7** [comment]
Yeah. Chrome OS does have some very well-planned features, and you have to credit the 3-second boot time. But there is a little too much vendor lockdown.

**Comment by kettle-7** [comment]
> DO tell me about your desktop environment.

![C'est mon environment, en français](https://user-images.githubusercontent.com/64388266/124344744-3e7b4280-dc28-11eb-8913-ddd8136f9808.png)
Some icons are missing in this picture: it used hardcoded paths to the Breeze icon theme which comes with KDE.

The window manager is `marco`, made by MATE, and the desktop icons and file manager are drawn by Çaja, also made by MATE.

**Comment by kettle-7** [comment]
Oh yes, and the dock part of the taskbar never worked.

**Comment by kettle-7** [comment]
So, if we're making a desktop environment, it will NOT be based on THAT.

**Comment by jhjacobs81** [comment]
Well, ive seen uglier DE’s in the world ;-) 

lets pitch arround some idea’s first.. i’ll ask my friend to look into whats needed for crowdfunding or whatever

**Comment by kettle-7** [comment]
> crowdfunding

What would we need money for? It's not like we're hiring staff, or a server backend!

**Comment by jhjacobs81** [comment]
True.. but if it gets lift, it would benefit from some fulltime staff ;-)

**Comment by kettle-7** [comment]
I'm working on a Qt theme for it, with nice gradients and no flatness.

**Comment by jhjacobs81** [comment]
Much like ODE??

**Comment by kettle-7** [comment]
What's ODE? A quick Google search shows a restaurant, a lamp shop, and a bunch of other businesses.

I'll upload a picture.

**Comment by jhjacobs81** [comment]
That would be Omega Desktop Environment ;-)

**Comment by kettle-7** [comment]
Oh right ;~)

**Comment by jhjacobs81** [comment]
;-) i thought it a cool name :)

**Comment by kettle-7** [comment]
![one](https://user-images.githubusercontent.com/64388266/124988021-99ca8c00-e091-11eb-871e-afef3058febe.png)

I haven't figured out those checkboxes, but these ones work:
![hell-oh](https://user-images.githubusercontent.com/64388266/124988390-09d91200-e092-11eb-95e7-5403d4baf7e8.png)

And then for some reason the combo-box dropdown menus are transparent...

**Comment by jhjacobs81** [comment]
This is Ubuntu?

**Comment by kettle-7** [comment]
Pop!_OS with MATE :)

**Comment by kettle-7** [comment]
Ubuntu looks like this:

![Screenshot from 2021-07-09 13-38-22](https://user-images.githubusercontent.com/64388266/125010622-00b06b00-e0bb-11eb-9c8b-98e34693e009.png)

**Comment by kettle-7** [comment]
I've been thinking about how Unity merges the titlebar and menubar of maximised windows, and I feel like we should do the same.

And hide the dock.

**Comment by kettle-7** [comment]
Do you have a personal preference for desktop environment? Cinnamon? MATE? GNOME? Xfce? If not, then LXQt or KDE?

I'm tweaking the Arch build script for Omega development, and I'll add a display manager.

**Comment by kettle-7** [comment]
I'll go LXQt for now, so that Qt gets installed properly, but we don't end up with piles of bloat everywhere.

442 packages later...

**Comment by jhjacobs81** [comment]
Any DE that we could make work on a desktop screen and a mobile screen ;-)

**Comment by kettle-7** [comment]
How about Debian... Arch is being cryptic about my wlan driver.

The driver is loaded, the firmware is loaded, but it *still* doesn't show up in /sys/class/net.

Debian never does this to me.

**Comment by kettle-7** [comment]
> building something Debian based.

It's definitely worth the extra 5 seconds of boot time. If we can get Hibernation working properly then we might just beat Chrome OS there.

Also bgrt Plymouth theme is a must.

**Comment by jhjacobs81** [comment]
I love Debian ;-)

**Comment by kettle-7** [comment]
Yeah. I've only used it once, for a couple of experiments and to see how it handles, on a 6GB VM with 6GB of RAM. But I have used plenty of Debian derivatives, like LMDE and Ubuntu.

I'll see how I can slim down systemd to make it faster.

Then there's Gentoo, I gave up trying to install it after about a month :)

**Comment by kettle-7** [comment]
It's plenty fast, it just needs a decent processor, and to not have the host Ubuntu hogging all of the CPU!

**Comment by jhjacobs81** [comment]
Huh?

**Comment by kettle-7** [comment]
I was worried about boot times, but they're pretty good. VMs are naturally slow.
